soccer/optimization/Gradient1DConfig.hpp

```diff
@@ -71,8 +68,7 @@ class Gradient1DConfig {
           maxValue(maxValue),
           maxThresh(maxThresh) {}
 
-    std::tuple<float, float>(*f)(float, FunctionArgs*);
-    FunctionArgs* args;
+    std::function<std::tuple<float, float>(float)>* f;
```

Comment:
Yeah, It's a pointer to a function pointer container. I might be over thinking this, but I was worried about the amount of data it may have to copy when being passed around.

When std::function wraps just a normal function pointer, it doesn't use that much space, like ~12 bytes on the stack. When you wrap the results of a std::bind or a lambda function, you have to worry about all the passed in arguments. For both, all the arguments are copied into the std::function. 

Additionally, std::function only has ~4-8 bytes of storage for this data. If you have anymore, it transfers everything to the heap.

For my use-case, it's anywhere from 4 - 26 copies depending on the amount of robots (It can even be more in some cases). My std::function size is around ~40 bytes (Assuming 32 bit system).

This works fine for simple arguments, but I was worried that if someone decides to throw in the robots vector or something bigger, it may cause a significant overhead that would be a pain to refactor later on.